app/assets/javascripts/discourse/tests/integration/components/d-editor-test.js

```diff
@@ -989,16 +976,33 @@ third line`
     }
   );
 
+  // Smart list functionality relies on beforeinput, which QUnit does not send with
+  // `typeIn` synthetic events. We need to send it ourselves manually along with `input`.
+  // Not ideal, but gets the job done.
+  //
+  // c.f. https://github.com/emberjs/ember-test-helpers/blob/master/API.md#typein and
+  // https://github.com/emberjs/ember-test-helpers/issues/1336
+  function triggerEnter(textarea) {
+    let event = new Event("beforeinput");
+    event.inputType = "insertLineBreak";
+    textarea.dispatchEvent(event);
+
+    event = new Event("input");
+    event.inputType = "insertText";
+    event.data = "\n";
+    textarea.dispatchEvent(event);
+  }
```

Comment:
Should this use: https://github.com/emberjs/ember-test-helpers/blob/master/API.md#triggerevent

So we can do async/await on triggerEnter